java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightServer.java

```diff
@@ -245,7 +254,21 @@ public FlightServer build() {
       }
 
       if (certChain != null) {
-        builder.useTransportSecurity(certChain, key);
+        SslContextBuilder sslContextBuilder = GrpcSslContexts
+                .forServer(certChain, key);
+
+        if (mTlsCACert != null) {
+          sslContextBuilder
+                  .clientAuth(ClientAuth.REQUIRE)
+                  .trustManager(mTlsCACert);
+        }
+        try {
+          sslContext = sslContextBuilder.build();
+        } catch (SSLException e) {
+          throw new RuntimeException(e);
+        }
```

Comment:
https://netty.io/4.1/api/io/netty/handler/ssl/SslContextBuilder.html#trustManager-java.io.InputStream-

```suggestion
        try {
          sslContext = sslContextBuilder.build();
        } catch (SSLException e) {
          throw new RuntimeException(e);
        } finally {
          if (mTlsCACert != null) {
            mTlsCACert.close();
          }
          certChain.close();
          key.close();
        }
```